[IMP] accounting/l10n_co: Update documentation for DIAN module (#11315)
odoo/documentation: content/applications/finance/fiscal_localizations/colombia.rst

========

.. |DIAN| replace:: :abbr:`DIAN (Dirección de Impuestos y Aduanas Nacionales)`
.. |NIT| replace:: :abbr:`NIT (El Número de Identificación Tributaria)`
.. _dian-certification: https://micrositios.dian.gov.co/sistema-de-facturacion-electronica/proceso-de-registro-y-habilitacion-como-facturador-electronico/

Odoo's Colombian localization package provides accounting, fiscal, and legal features for databases
in Colombia - such as chart of accounts, taxes, and electronic invoicing. The localization has the
following `prerequisites
<https://micrositios.dian.gov.co/sistema-de-facturacion-electronica/que-requieres-para-factura-electronicamente/>`_
when using the `DIAN Own Software
<https://micrositios.dian.gov.co/sistema-de-facturacion-electronica/como-puedes-facturar-electronicamente/>`_
solution with Odoo:

- Be registered in the `RUT
  <https://www.dian.gov.co/tramitesservicios/tramites-y-servicios/tributarios/Paginas/RUT.aspx>`_
  (Registro Único Tributario) with a valid |NIT|.
- Have a valid digital signature certificate `approved by the ONAC
  <https://onac.org.co/directorio-de-acreditados/>`_.

> - `Register and get enabled <dian-certification>`_ by completing the certification process required

Author: ```suggestion
- `Register and get enabled <https://micrositios.dian.gov.co/sistema-de-facturacion-electronica/proceso-de-registro-y-habilitacion-como-facturador-electronico/>`_ by completing the certification process required
```

Author: @samueljlieber added this link so it can work and take you to the defined website

  by the |DIAN|.

.. seealso::
   - For more information on how to complete the certification process for the |DIAN| module, review

>      the following `webinar <dian-certification>`_.

Author: ```suggestion
     the following `webinar <https://www.youtube.com/watch?v=l0G6iDc7NQA>`_.
```
@samueljlieber Also here, the link needed to be redirected to

- :guilabel:`Commercial Name`: If the company uses a specific commercial name, and it needs to be
  displayed in the invoice.

> 

Author: ```suggestion
.. tip::
   The data configured in the :guilabel:`Fiscal Information` section is printed in fiscal valid PDF reports.

.. image:: colombia/fiscal-data-in-fiscal-pdf-con.png
   :alt: Fiscal information printed in PDF.
```
@samueljlieber this is my proposal after deprecating the "Report Configuration" section that was only aimed for Carvajal. (Added the new image to the same [drive folder](https://drive.google.com/drive/folders/11cXF1gXxwws_P9hUslnSJcfjheEbBJZN))

with |DIAN|'s web service. To do so, navigate to :menuselection:`Accounting --> Configuration -->
Settings` and scroll to the :guilabel:`Colombian Electronic Invoicing` section. Then, complete the
following configuration:

- Select :guilabel:`DIAN: Free Service` as the :guilabel:`Electronic Invoicing Provider`.
- Configure the :guilabel:`Operation Modes` for the respective types of documents (*electronic
  invoices* or *support documents*) to be generated from Odoo:

  - :guilabel:`Software Mode`: the type of document to be generated with the operation mode.
  - :guilabel:`Software ID`: the ID generated by |DIAN| for the specific operation mode.
  - :guilabel:`Software PIN`: the PIN selected in the operation mode configuration in the |DIAN|
    portal.
  - :guilabel:`Testing ID`: the testing ID generated by |DIAN| and obtained after testing the
    operation mode.

- Configure the available :guilabel:`Certificates` to sign your electronic documents:

  - :guilabel:`Name`: the name of the certificate.
  - :guilabel:`Certificate`: upload the certificate file in PEM format. After uploading a valid
    certificate file, the following fields display:

>     - :guilabel:`Certificate Password`: the password to decrypt the PKS file.

Author: ```suggestion

```
@samueljlieber this passwork is not needed for Colombia, since the certificates for this country can only be decoded by a PEM key file

>     - :guilabel:`Private Key`: select or create a new a private key.

Author: ```suggestion
    - :guilabel:`Private Key`: select or create a new a private key. When creating it, upload the key PEM file related to your certificate into the :guilabel:`Key file` field.
```
@samueljlieber detailing a bit more this process, if you have feedback let me know (screenshot below of the process)

![image](https://github.com/user-attachments/assets/69c6086c-6f9b-42ec-842a-54a8ec551dcf)

>     - :guilabel:`Public Key`: Set a public key to be used if the key contained in the certificate

Author: ```suggestion

```
@samueljlieber similar, there cannot be a Public Key

>       file is invalid.

Reviewer: I detailed the **Certificates** section with more instruction for each field, can you please check these instructions? :)

Author: I left some comments!

the :guilabel:`Colombian Electronic Invoicing` section:

- **Certification environment**: This environment is useful to pass the |DIAN| certification
  process and obtain the *Enabled* status to invoice from Odoo. To activate it, tick both the
  :guilabel:`Test Environment` and the :guilabel:`Activate the certification process` checkboxes.

- **Testing environment**: This environment allows reproducing electronic invoicing flows
  and validations in the |DIAN| testing portal. To activate it, tick only the :guilabel:`Test
  Environment` checkbox.

- **Production environment**: Activate production databases to generate valid electronic documents.
  To activate it, untick both the :guilabel:`Test Environment` and the :guilabel:`Activate the
  certification process` checkboxes.

.. seealso::
   For electronic invoicing configurations using the Carvajal solution, review the following video:
   `Configuración de Facturación Electrónica - Localización de Colombia
   <https://www.youtube.com/watch?v=bzweMwTEbfY&list=PL1-aSABtP6ABxZshems3snMjx7bj_7ZsZ&index=3>`_.

Report data configuration
-------------------------

.. Only for Carvajal config ⬇️

> 

Reviewer: In my testing, the "Report Configuration" is only visible in the **Colombian Electronic Invoicing** section when **Carvajal** is selected as the **Electronic Invoicing Provider**. Can you please double check this? :)

Author: @samueljlieber this is totally true! In fact this `Report data configuration` section should be deprecated completely. I will add "Tip" and possibly a screenshot in the `Company configuration` section. Since now these fields are used for the fiscal information of the PDF

    operation mode.

- Configure the available :guilabel:`Certificates` to sign your electronic documents:

  - :guilabel:`Name`: the name of the certificate.
  - :guilabel:`Certificate`: upload the certificate file in PEM format. After uploading a valid
    certificate file, the following fields display:

    - :guilabel:`Private Key`: Select an existing or create a new a private key by typing in a name
      for the key, and select :guilabel:`Create and edit...` to display the :guilabel:`Create
      Private Key` wizard. Then, upload a valid :guilabel:`Key file` and :guilabel:`Save & Close`.

.. image:: colombia/dian-credentials-configuration.png
   :alt: Colombian electronic invoicing credentials configured.

.. note::
   In a multi-company database, it is possible to have a different certificate for each company.

.. _colombia/dian-environment-config:

|DIAN| environment configuration
--------------------------------

The |DIAN| electronic invoicing module offers three different |DIAN| environments to connect with,
each can be enabled from :menuselection:`Accounting --> Configuration --> Settings` and scroll to
the :guilabel:`Colombian Electronic Invoicing` section:

> - **Certification environment**: This environment is useful to pass the |DIAN| certification

Reviewer: Hi @afma-odoo! I disagree that these environment titles should use a :guilabel: because these titles are not used anywhere in the UI, rather, they are titles used to describe a specific configuration of fields in the UI.

For example, ``` :guilabel:`Test environment` ``` should have a :guilabel: since it is UI text. Let me know if you agree? :)

<img width="985" alt="Screenshot 2024-11-08 at 3 32 32 PM" src="https://github.com/user-attachments/assets/0b80049f-14c2-4e66-afc3-e2663a5a6344">

> Partner

Author: Agree with this improvement

~~~~~~~

> Partner contacts can be created in the **Contacts** app. To do so, navigate to

Author: @samueljlieber could you take a look at this? :)

- :guilabel:`Gran Contribuyente`: If the company is *Gran Contribuyente* this option should be
  selected.

> - :guilabel:`Fiscal Regimen`: Select the tribute name for the company (`IVA`, `INC`, `IVA e INC`, or

Author: @samueljlieber do you have an insight on this? Just looking for the best practice here, since these are options the user will select from

> - :guilabel:`Fiscal Regimen`: Select the tribute name for the company (`IVA`, `INC`, `IVA e INC`, or

Reviewer: Yea I agree with @afma-odoo :) since these values are part of a fixed drop-down menu, they should be considered interactive UI text and therefore need a :guilabel:

<img width="571" alt="Screenshot 2024-11-08 at 3 46 16 PM" src="https://github.com/user-attachments/assets/9f651d82-60e5-4cee-94c2-2c0f04c98afd">

However, conversely I'd argue we can use literal formatting (\` \`) for fields that take user input, such as the **Obligaciones y Responsabilidades** field:

<img width="364" alt="Screenshot 2024-11-08 at 3 50 25 PM" src="https://github.com/user-attachments/assets/d652e5a0-67d4-4264-8e55-6e2efb71611d">

Open to thoughts!

> - :guilabel:`Fiscal Regimen`: Select the tribute name for the company (`IVA`, `INC`, `IVA e INC`, or

Reviewer 2: @samueljlieber Literal should be used when there's no option to select. In this case, there are options so it should be :guilabel:

   The functional workflow taking place before an invoice validation does **not** alter the main
   changes introduced with the electronic invoice.

Electronic invoices are generated and sent to both the |DIAN| and the customer. These documents can
be created from the sales order or manually generated. To create a new invoice, go to
:menuselection:`Accounting --> Costumers --> Invoices`, and select :guilabel:`Create`. On the
invoice form, configure the following fields:

- :guilabel:`Customer`: customer's information.
- :guilabel:`Journal`: journal used for electronic invoices.
- :guilabel:`Electronic Invoice Type`: Select the type of document. By default, :guilabel:`Factura
  de Venta` is selected.
- :guilabel:`Invoice Lines`: Specify the products with the correct taxes.

.. important::
   When creating the first invoice related to an electronic invoicing journal, it is required to
   manually change the *sequence* of the invoice to the |DIAN| format: `Prefix + Sequence`.

>    For example, format the sequence from `SETP1/2024/00001` to `SETP1`.

Author: @auva-odoo for the example mentioned.
In Odoo, by standard, invoice sequences use the short code from the journal, the year and a chronological sequence (sepereated by "/" symbols). This format is not accepted by the DIAN in Colombia.

This is why it is very important that at the moment of creating the first invoice (where Odoo lets you manually change the sequence) the user changes it to a format that will be accepted by the DIAN which is only the "Prefix + Sequence".

`SETP1/2024/00001` to `SETP1`

To add more information, companies in Colombia need to go through a process where they request their desired "Prefix" to the DIAN, and it needs to be validated: https://www.youtube.com/watch?v=YM_tJgdj4_c

Author: ```suggestion
   For example, format the sequence from `SETP/2024/00001` to `SETP1`.
```
In fact if we make this change it should be clearer

.. important::
   When creating the first invoice related to an electronic invoicing journal, it is required to
   manually change the *sequence* of the invoice to the |DIAN| format: `Prefix + Sequence`.

   For example, format the sequence from `SETP1/2024/00001` to `SETP1`.

When done, click :guilabel:`Confirm`.

.. _colombia/send-electroinic-invoice:

Sending electronic invoices
~~~~~~~~~~~~~~~~~~~~~~~~~~~

After the :ref:`invoice confirmation <colombia/invoice-creation>`, click on the :guilabel:`Print &
Send` button. In the wizard that appears, make sure to enable the :guilabel:`DIAN` and
:guilabel:`Email` checkboxes to send an XML to the |DIAN| web service and the validated invoice to
your client fiscal email. After that, click on the :guilabel:`Print & Send` button:

- The XML document is created
- CUFE is generated
- The XML is processed synchronously by the |DIAN|.
- If accepted, the file is displayed in the chatter and also the email to the client with the
  corresponding :file:`.zip` file.

.. image:: colombia/zip-xml-chatter-colombia.png
   :alt: EDI documents available in the chatter.

The :guilabel:`DIAN` tab now displays the following:

> - :guilabel:`Signed Date`: timestamp recorded of the XML creation.
> - :guilabel:`Status`: The status result obtained in the |DIAN| response. If the invoice was
>   rejected, the error messages can be seen here.
> - :guilabel:`Testing Environment`: This checkbox will let us know if the document sent was delivered
>   to the |DIAN| testing environment.
> - :guilabel:`Certification Process`: This checkbox will let us know if the document was sent as part
>   of the certification process with the |DIAN|.
> - :guilabel:`Download button`: With this button, it is possible to download the sent XML file, even
>   if the |DIAN| result was rejected.
> - :guilabel:`Fetch Attached Document button`: With this button, it is possible to download the
>   generated attached document file, even if the |DIAN|.

Reviewer 2: ```suggestion
- :guilabel:`Signed Date`: timestamp recorded of the XML creation
- :guilabel:`Status`: Status result obtained in the |DIAN| response. If the invoice was
  rejected, the error messages can be seen here.
- :guilabel:`Testing Environment`: To know if the document sent was delivered to the |DIAN| testing
  environment.
- :guilabel:`Certification Process`: To know if the document was sent as part of the certification
  process with the |DIAN|.
- :guilabel:`Download`: To download the sent XML file, even if the |DIAN| result was 
  rejected.
- :guilabel:`Fetch Attached Document`: To download the generated attached document file, 
  even if the |DIAN|.
```
To simplify the explanations and be more direct.

> - :guilabel:`Fetch Attached Document button`: With this button, it is possible to download the
>   generated attached document file, even if the |DIAN|.

Author: ```suggestion
- :guilabel:`Fetch Attached Document button`: With this button, it is possible to download the
  generated attached document file, also included in the delivered :file:`.zip` file to the client.
```

-------------

During the XML validation, the most common errors are related to missing master data (*Contact Tax
ID*, *Address*, *Products*, *Taxes*). In such cases, rejections from the |DIAN| are prevented by

> showing validation error messages before being able to send the documents to the |DIAN|.

Author: @afma-odoo @samueljlieber I think we can remove this image if needed

> ID*, *Address*, *Products*, *Taxes*). In such cases, error messages are shown in the chatter after

Author: ```suggestion
ID*, *Address*, *Products*, *Taxes*). In such cases, rejections from the DIAN are prevented by showing validation error messages before being able to send the documents to the DIAN
```

> updating the electronic invoice status.

Author: ```suggestion

```

Commerce (ICA) tax. The report can be found under :menuselection:`Accounting --> Reporting -->
Colombian Statements --> Certificado de Retención en ICA`.

> .. image:: colombia/retencion-ica-dian.png

Author: @afma-odoo the images are helpful to show how the user must "open" the partners detail in the report to see the desired information for Colombian legal requirements.

Do you think explaining this by text would be better? 

CC @samueljlieber

> .. image:: colombia/retencion-ica-dian.png

Reviewer: We should prioritize text instructions over images, so I agree we need to add text describing how to open the partner detail in the report. I do think the images help illustrate each report, although not entirely necessary. 

I will add text instructions and leave images for now, please let me know if you'd like these 3 images to be removed!

Colombian Statements --> Certificado de Retención en ICA`.

> .. image:: colombia/ica-report.png

Author: @samueljlieber Change to `retencion-ica-dian.png` already in [Google Drive folder](https://drive.google.com/drive/folders/11cXF1gXxwws_P9hUslnSJcfjheEbBJZN) :)

de Retención en IVA`.

> .. image:: colombia/iva-report.png

Author: @samueljlieber Change to `retencion-iva-dian.png` already in [Google Drive folder](https://drive.google.com/drive/folders/11cXF1gXxwws_P9hUslnSJcfjheEbBJZN) :)

Retención en Fuente`.

> .. image:: colombia/fuente-report.png

Author: @samueljlieber Change to `retencion-fuente-dian.png` already in [Google Drive folder](https://drive.google.com/drive/folders/11cXF1gXxwws_P9hUslnSJcfjheEbBJZN) :)
